onnxruntime/core/graph/featurizers_ops/featurizers_defs.cc

```diff
@@ -408,6 +410,133 @@ void RegisterDateTimeFeaturizerVer1() {
+  //                 RW[0][0] == na, LL[0][0] == na, LL[1][0] == na;   na's found, nothing to output
+  //                 RW[0][1] == na, LL[0][1] == na, LL[1][1] == na;   na's found, nothing to output
+  //                 RW[0][2] == na, LL[0][2] == na, LL[1][2] == na;   na's found, nothing to output
+
+  //               ...
+
+  //               4:
+  //                 RW[0][0] == 1, LL[0][0] == A, LL[1][0] == D;      no na's found - OUTPUT GENERATED (1, A, D)
+  //                 RW[0][1] == 2, LL[0][1] == B, LL[1][1] == na;     na's found, nothing to output
+  //                 RW[0][2] == 3, LL[0][2] == C, LL[1][2] == na;     na's found, nothing to output
+
+  //               5:
+  //                 RW[0][0] == 1, LL[0][0] == A, LL[1][0] == D;      no na's found - OUTPUT GENERATED (1, A, D)
+  //                 RW[0][1] == 2, LL[0][1] == B, LL[1][1] == na;     na's found, nothing to output
+  //                 RW[0][2] == 3, LL[0][2] == C, LL[1][2] == F;      no na's found - OUTPUT GENERATED (3, C, F)
+  //)DOC";
+
+  MS_FEATURIZERS_OPERATOR_SCHEMA(ForecastingPivotTransformer)
+      .SinceVersion(1)
+      .SetDomain(kMSFeaturizersDomain)
+      .Input(
+          0,
+          "State",
+          "State generated during training that is used for prediction",
+          "T0")
+      .Input(
+          1,
+          "Inputs",
+          "Variadic number of Input containing tensors of different size",
+          "T",
+          ONNX_NAMESPACE::OpSchema::FormalParameterOption::Variadic)
+      .Output(
+          0,
+          "Output",
+          "No information is available",
+          "T")
+      .TypeConstraint(
+          "T0",
+          {"tensor(uint8)"},
+          "No information is available")
+      .TypeConstraint(
+          "T",
+          {"tensor(float)", "tensor(double)"},
+          "No information is available")
+      .TypeAndShapeInferenceFunction(
+          [](ONNX_NAMESPACE::InferenceContext& ctx) {
+            auto input_elem_type = ctx.getInputType(1)->tensor_type().elem_type();
+            if (input_elem_type == ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
+              propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_FLOAT, 0);
+            } else if (input_elem_type == ONNX_NAMESPACE::TensorProto_DataType_DOUBLE) {
+              propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_DOUBLE, 0);
+            } else {
+              fail_type_inference("input 1 is expected to have an accepted type");
+            }
+            if (hasInputShape(ctx, 1)) {
+              const auto& input_shape = getInputShape(ctx, 1);
+              if (input_shape.dim_size() != 3) {
+                fail_shape_inference("Expecting Inputs to have 3 dimensions");
+              }
+              ONNX_NAMESPACE::TensorShapeProto shape;
+              shape.add_dim();
+              shape.add_dim();
+              ONNX_NAMESPACE::updateOutputShape(ctx, 0, shape);
```

Comment:
This piece of code is not dependent on whether Input 1 has a shape. It should not be under if()